client/src/Entity/CasRec.php

```diff
@@ -0,0 +1,290 @@
+        // Prof
+        [true, self::REALM_PROF, ['p3', 'p3g'], 'opg103', Report::PROF_PFA_LOW_ASSETS_TYPE],
+        // @deprecated (DDPB-2044)
+        [true, self::REALM_PROF, ['l3', 'l3g', 'a3'], 'opg103', Report::PROF_PFA_LOW_ASSETS_TYPE],
+        [true, self::REALM_PROF, ['p2', 'p2a'], 'opg102', Report::PROF_PFA_HIGH_ASSETS_TYPE],
+        // @deprecated (DDPB-2044)
+        [true, self::REALM_PROF, ['l3', 'l3g', 'a3'], 'opg102', Report::PROF_PFA_HIGH_ASSETS_TYPE],
+        [true, self::REALM_PROF, ['hw'], '', Report::PROF_HW_TYPE],
+        [true, self::REALM_PROF, ['hw'], 'opg103', Report::PROF_COMBINED_LOW_ASSETS],
+        [true, self::REALM_PROF, ['hw'], 'opg102', Report::PROF_COMBINED_HIGH_ASSETS],
+    ];
+
+    /**
+     * @var int
+     */
+    private $id;
+
+    /**
+     * @var string
+     *
+     * @JMS\Type("string")
+     *
+     * @Assert\NotBlank()
+     */
+    private $caseNumber;
+
+    /**
+     * @var string
+     *
+     * @JMS\Type("string")
+     *
+     * @Assert\NotBlank()
+     */
+    private $clientLastname;
+
+    /**
+     * @var string
+     *
+     * @JMS\Type("string")
+     *
+     * @Assert\NotBlank()
+     */
+    private $deputyNo;
+
+    /**
+     * @var string
+     *
+     * @JMS\Type("string")
+     *
+     * @Assert\NotBlank()
+     */
+    private $deputySurname;
+
+    /**
+     * @var string
+     *
+     * @JMS\Type("string")
+     *
+     * @Assert\Length(min=2, max=10, minMessage="postcode too short", maxMessage="postcode too long" )
+     */
+    private $deputyPostCode;
+
+    /**
+     * @var string OPG102|OPG103|empty string
+     *
+     * @JMS\Type("string")
+     */
+    private $typeOfReport;
+
+    /**
+     * @var string A2|C1|HW|L2|L2A|L3|L3G|P2A|PGA|PGC|S1A|S1N|empty
+     *
+     * typeOfReport=OPG103 only have
+     *
+     * @JMS\Type("string")
+     */
+    private $corref;
+
+    /**
+     * @JMS\Type("string")
+     */
+    private $otherColumns;
+
+    /**
+     * @var \DateTime
+     *
+     * @JMS\Type("DateTime<'Y-m-d H:i:s'>")
+     */
+    private $createdAt;
+
+    /**
+     * @var \DateTime
+     *
+     * @JMS\Type("DateTime<'Y-m-d H:i:s'>")
+     */
+    private $updatedAt;
+
+    /**
+     * @var string
+     */
+    private $source;
+
+    /**
+     * @var \DateTime
+     */
+    private $orderDate;
+
+    public function __construct()
+    {
+    }
+
+    public function getCaseNumber()
+    {
+        return $this->caseNumber;
+    }
+
+    public function getClientLastname()
+    {
+        return $this->clientLastname;
+    }
+
+    public function getDeputyNo()
+    {
+        return $this->deputyNo;
+    }
+
+    public function getDeputySurname()
+    {
+        return $this->deputySurname;
+    }
+
+    public function getDeputyPostCode()
+    {
+        return $this->deputyPostCode;
+    }
+
+    /**
+     * @return string
+     */
+    public function getTypeOfReport()
```

Comment:
Should we be type hinting the end of the function with `: string` if we are adding the DocBlock for it as well?

Reply:
Considering we type hint it inside the `CSVUploadedEvent` then for consistency I think we should type hint in this file also for the functions

Reply:
Yeah I don't think I've been consistent with the way in which I type hint.
I think the PSR-12 removes the DocBlock if we type hint in the function declaration.

Before I merge, I'll try to make the type hinting consistent, and experiment with what the formatting pre-hook does.